src/NServiceBus.Core/Pipeline/PipelineSettings.cs

```diff
@@ -90,28 +90,27 @@ public StepRegistrationSequence Register(Type behavior, string description)
         /// <param name="stepId">The identifier of the new step to add.</param>
         /// <param name="behavior">The <see cref="Behavior{TContext}" /> to execute.</param>
         /// <param name="description">The description of the behavior.</param>
-        public StepRegistrationSequence Register(string stepId, Type behavior, string description)
+        public void Register(string stepId, Type behavior, string description)
         {
             BehaviorTypeChecker.ThrowIfInvalid(behavior, nameof(behavior));
 
             Guard.AgainstNullAndEmpty(nameof(stepId), stepId);
             Guard.AgainstNullAndEmpty(nameof(description), description);
 
             AddStep(RegisterStep.Create(stepId, behavior, description));
```

Comment:
Given that the primary reason for `AddStep()` as a separate method was to be able to pass it to `StepRegistrationSequence`, do you still see value in keeping it around, or should `modifications.Additions.Add(step);` just be inlined?